lte/gateway/python/magma/subscriberdb/client.py

```diff
@@ -77,32 +78,47 @@ def __init__(
         self._grpc_client_manager = grpc_client_manager
 
     async def _run(self) -> None:
-        subscribers = await self._get_subscribers()
-        if subscribers is not None:
+        subscribers, no_updates = await self._get_subscribers()
+        # No need to update local db if it's already up-to-date.
+        if (subscribers is not None) and (not no_updates):
             self._process_subscribers(subscribers)
```

Comment:
- Always try to make boolean logic as simple as possible. One tool in your toolset: https://en.wikipedia.org/wiki/De_Morgan%27s_laws
- Early returns are nice, especially when they allow for simpler boolean logic

```suggestion
        if subscribers is None or no_updates:
            return
        self._process_subscribers(subscribers)
```